src/util/encode-url-for-id.js

```diff
@@ -22,21 +22,27 @@ const normalizationOpts = {
  */
 function applyQueryParamsRules(url) {
     const parsed = new URL(url)
-    const rules = queryParamRules.get(parsed.hostname)
+    const rulesObj = queryParamRules.get(parsed.hostname)
 
     // Base case; domain doesn't have any special normalization rules
-    if (!rules) {
+    if (!rulesObj) {
         return url
     }
 
     // Remove all query params that don't appear in special rules
-    const rulesSet = new Set(rules)
-    for (const param of parsed.searchParams.keys()) {
-        if (!rulesSet.has(param)) {
-            parsed.searchParams.delete(param)
+    const rulesSet = new Set(rulesObj.rules)
+    const rulesType = rulesObj.type
+    if (rulesType == 'keep') {
```

Comment:
To keep it simple as possible, I think the only thing that needs to change in this file is the condition within the `for...of` loop. It should either be `!rulesSet.has(param)` for `'keep'` strategy or `rulesSet.has(param)` for `'remove'`. This could just be a single line before the existing `if`

Also be careful of the abstract equality operator `==`. Here the strict equality op is more appropriate `===`